File: AndroidBasics/calculator/src/main/java/pan/alexander/calculator/di/ApplicationComponent.java
package pan.alexander.calculator.di;

import javax.inject.Singleton;

import dagger.Component;
import pan.alexander.calculator.data.settings.SettingsRepositoryDao;
import pan.alexander.calculator.data.database.HistoryDao;
import pan.alexander.calculator.domain.Calculator;
import pan.alexander.calculator.domain.MainInteractor;

@Singleton
@Component(modules = {SettingsModule.class, RepositoryModule.class, RoomModule.class})
public interface ApplicationComponent {
    HistoryDao getHistoryDao();
    SettingsRepositoryDao getSettingsRepositoryDao();
    MainInteractor getMainInteractor();
    Calculator getCalculator();
}

File: PopularLibs/logintask/src/main/java/pan/alexander/logintask/di/ApplicationComponent.kt
package pan.alexander.logintask.di

import dagger.Component
import pan.alexander.logintask.ui.login.LoginActivity
import javax.inject.Singleton

@Singleton
@Component(modules = [RepositoryModule::class, CoroutinesModule::class, PresentersModule::class])
interface ApplicationComponent {
    fun inject(loginActivity: LoginActivity)
}
